react/states_and_effects/more_on_state.md

```diff
@@ -201,13 +204,16 @@ This pattern is extremely useful wherever you need user input i.e. typing in a t
 This section contains questions for you to check your understanding of this lesson on your own. If you’re having trouble answering a question, click it and review the material it links to.
 
 - [What should you keep in mind while declaring state?](#how-to-structure-state)
-- [What does "state is immutable" mean?](#state-is-immutable)
+- [Why should we always use `setState` to update our state?](#state-should-not-be-mutated)
 - [What does "state as a snapshot" mean?](#how-state-updates)
 - [What's the difference between passing a value vs a callback to the `setState` function?](#state-updater-functions)
+- [Why should we always provide a new Object to `setState`?](#objects-and-arrays-in-state)
 - [Why would you want to control a component?](#controlled-components)
 
 ### Additional resources
 
 This section contains helpful links to related content. It isn’t required, so consider it supplemental.
 
 - We intentionally skipped some of the articles pertaining to state from the React documentation. We are confident that the concepts covered thus far will equip you with enough knowledge to tackle the projects to come in the course. Though if you want to delve into the topic further, you can read all of the articles in the [Adding Interactivity](https://react.dev/learn/adding-interactivity) and the [Managing State](https://react.dev/learn/managing-state) section.
+- React tip [Why is mutating state not recommended in React?](https://react.dev/learn/updating-objects-in-state#why-is-mutating-state-not-recommended-in-react)
+- React tip [is using an updater always preferred?](https://react.dev/reference/react/useState#is-using-an-updater-always-preferred)
```

Comment:
I think the first additional resource covers the next two articles. I'm fine with removing the other two additional resources. I'll leave the decision to you. Though if we keep those, I'd suggest wording changes:

```suggestion
- We intentionally skipped some of the articles pertaining to state from the React documentation. We are confident that the concepts covered thus far will equip you with enough knowledge to tackle the projects to come in the course. Though if you want to delve into the topic further, you can read all of the articles in the [Adding Interactivity](https://react.dev/learn/adding-interactivity) and the [Managing State](https://react.dev/learn/managing-state) section.
- [Why is mutating state not recommended in React?](https://react.dev/learn/updating-objects-in-state#why-is-mutating-state-not-recommended-in-react)
-  [Is using a state updater always preferred?](https://react.dev/reference/react/useState#is-using-an-updater-always-preferred)
```